File: app/src/main/java/com/przemolab/oknotifier/modules/OpenKattisServiceModule.java
package com.przemolab.oknotifier.modules;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class OpenKattisServiceModule {

    @Provides
    @Singleton
    public OpenKattisService provideOpenKattisService() {
        return new OpenKattisService();
    }
}

File: app/src/main/java/com/przemolab/oknotifier/modules/OpenKattisServiceModule.kt
package com.przemolab.oknotifier.modules

import com.przemolab.oknotifier.interfaces.IOpenKattisService
import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
open class OpenKattisServiceModule {

    @Provides
    @Singleton
    open fun provideOpenKattisService(): IOpenKattisService {
        return OpenKattisService()
    }
}
